guardrails/run/async_stream_runner.py

```diff
@@ -153,6 +153,10 @@ async def async_step(
                     validate_subschema=True,
                     stream=True,
                 )
+                # TODO why? how does it happen in the other places we handle streams
```

Comment:
@nichwch any insights?

Reply:
If you're asking about the None check on line 157, it might be a remnant of before we implemented generators all the way down. Before, it was possible for validation to return None of the validators haven't accumulated enough chunks to validate yet. For sync streaming, we've changed this so that the validation logic takes place in a generator and only emits results when enough chunks have been accumulated. 

Looking into the async streaming code, it looks like we never changed that to use generators for the validation logic, so validators still emit Nones before they have accumulated enough chunks